src/wasm-type.h

```diff
@@ -240,35 +227,119 @@ struct ResultType {
   std::string toString() const;
 };
 
+struct Tuple {
+  TypeList types;
+  Tuple() : types() {}
+  Tuple(std::initializer_list<Type> types) : types(types) {}
+  Tuple(const TypeList& types) : types(types) {}
+  Tuple(TypeList&& types) : types(std::move(types)) {}
+  bool operator==(const Tuple& other) const { return types == other.types; }
+  bool operator!=(const Tuple& other) const { return !(*this == other); }
+  std::string toString() const;
+};
+
 struct Signature {
   Type params;
   Type results;
   Signature() : params(Type::none), results(Type::none) {}
-  Signature(Type params, Type results) : params(params), results(results) {}
+  Signature(const Type& params, const Type& results)
+    : params(params), results(results) {}
```

Comment:
I think this can still take the `Type`s by value, since it's not an operator overload or anything like that.

```suggestion
  Signature(Type params, Type results) : params(params), results(results) {}
```